test/appendGitlabProv.test.js

```diff
@@ -0,0 +1,103 @@
+import { strictEqual } from 'assert'
+import namespace from '@rdfjs/namespace'
+import assertThrows from 'assert-throws-async'
+import getStream from 'get-stream'
+import { isDuplex } from 'isstream'
+import { describe, it } from 'mocha'
+import rdf from 'rdf-ext'
+import { Readable } from 'readable-stream'
+import appendGitlabProv from '../lib/appendGitlabProv.js'
+import { dcat } from '../lib/namespaces.js'
+import {
+  clearMockEnvironment, setMockEnvironment
+} from './support/gitlabEnvironment.js'
+
+const ex = namespace('http://example.org/')
+const typePredicate = rdf.namedNode(
+  'http://www.w3.org/1999/02/22-rdf-syntax-ns#type')
+
+describe('metadata.appendGitlabProv', () => {
+  it('should be a factory', () => {
+    strictEqual(typeof appendGitlabProv, 'function')
+  })
+
+  it('should throw an error if no argument is given', async () => {
+    await assertThrows(async () => {
+      await appendGitlabProv()
+    }, Error, /Needs subjectsWithClass as parameter/)
+  })
```

Comment:
On second thought, I think I would not throw but log a warning and continue as pass-through. Not sure though

Reply:
I think it needs to stop, so the developer can fix the pipeline. 

What I'm not sure about is if warnings should be shown if the pipeline is run outside GitLab.

Reply:
Makes sense. 'Gitlab detected' is ok, but this pull request only knows about Gitlab